eng/Workarounds.targets

```diff
@@ -1,10 +1,13 @@
 <!-- Use this file to workaround issues. List the issue tracking the item to fix so we can remove the workaround when the issue is resolved. -->
 <Project>
   <ItemGroup>
-    <!-- Reference base shared framework at incoming dependency flow version, not bundled sdk version. -->
+    <!--
+      Reference base shared framework at incoming dependency flow version, not bundled sdk version.
+      Apply this to all projects that target the default tfm (e.g. net6.0) or a rid-based variant of it (e.g. net6.0-windows)
+    -->
     <FrameworkReference
       Update="Microsoft.NETCore.App"
-      Condition=" '$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)' AND '$(TargetLatestDotNetRuntime)' != 'false' "
+      Condition=" (('$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)') OR $(TargetFramework.StartsWith('$(DefaultNetCoreTargetFramework)-'))) AND '$(TargetLatestDotNetRuntime)' != 'false' "
```

Comment:
FYI @dougbu

Reply:
Makes sense but TFM parsing isn't a great idea. Suggest using
``` msbuild
<PropertyGroup>
  <DefaultNetCoreTargetFrameworkIdentifier>$([MSBuild]::GetTargetFrameworkIdentifier('$(DefaultNetCoreTargetFramework)'))</DefaultNetCoreTargetFrameworkIdentifier>
  <DefaultNetCoreTargetFrameworkVersion>v$([MSBuild]::GetTargetFrameworkVersion('$(DefaultNetCoreTargetFramework)', 2))</DefaultNetCoreTargetFrameworkVersion>
</PropertyGroup>
```
then do exact comparisons of those properties against `$(TargetFrameworkIdentifier)` and `$(TargetFrameworkVersion)`.

Even better to put the property settings into eng/Versions.props right after setting `$(DefaultNetCoreTargetFramework)` because I suspect we'll need to use them more in the future.